### PR TITLE
KAFKA-2645: Document potentially breaking changes in the release note…

### PR DESCRIPTION
**Pull request by granthenke**
…s for 0.9.0

### CODE REVIEW

**Comment by granthenke** [comment]
This should address the documentation needed for #310 too.

File: docs/upgrade.html [hunk 1]
```diff
@@ -31,6 +31,23 @@
 
 Note: Bumping the protocol version and restarting can be done any time after the brokers were upgraded. It does not have to be immediately after.
 
+<h5>Potential breaking changes in 0.9.0.0</h5>
+
+<ul>
+    <li> Tools packaged under <em>org.apache.kafka.clients.tools.*</em> have been moved to <em>org.apache.kafka.tools.*</em>. All included scripts will still function as usual, only custom code directly importing these classes will be affected. </li>
+    <li> The default Kafka JVM performance options (KAFKA_JVM_PERFORMANCE_OPTS) have been changed in kafka-run-class.sh. </li>
+    <li> The kafka-topics.sh script (kafka.admin.TopicCommand) now exits with non-zero exit code on failure. </li>
```
Comment (guozhangwang) [inline docs/upgrade.html:40]: @junrao @gwenshap Do we need to include the third and forth item as breaking changes?

**Comment by onurkaraman** [comment]
The "auto.offset.reset" config options changed from ZookeeperConsumerConnector and KafkaConsumer.
ZookeeperConsumerConnector values: "smallest" or "largest"
KafkaConsumer values: "earliest", "latest", or "none"

**Comment by guozhangwang** [comment]
@onurkaraman We need to include this in the new consumer client docs, while the breaking changes here are related to brokers only.

File: docs/upgrade.html (hunk 1 above)
Comment (gwenshap) [inline docs/upgrade.html:40]: I suggest leave them here - its super minor and maybe breaks nothing, but better safe than sorry.

File: docs/upgrade.html [hunk 2]
```diff
@@ -31,6 +31,23 @@
...
+    <li> The kafka-topics.sh script (kafka.admin.TopicCommand) will now print a warning when topic names risk metric collisions due to the use of a '.' or '_' in the topic name, and error in the case of an actual collision. </li>
+    <li> By default all command line tools will print all logging messages to stderr instead of stout. </li>
```
Comment (gwenshap) [inline docs/upgrade.html:42]: Add something about JVM 1.6 no longer supported?

**Pull request closed without merging**